Tests/FoundationInternationalizationTests/Formatting/DateRelativeFormatStyleTests.swift

```diff
@@ -307,24 +307,88 @@ final class DateRelativeFormatStyleTests: XCTestCase {
     func testAutoupdatingCurrentChangesFormatResults() {
         let locale = Locale.autoupdatingCurrent
         let date = Date.now + 3600
-        
+
         // Get a formatted result from es-ES
         var prefs = LocalePreferences()
         prefs.languages = ["es-ES"]
         prefs.locale = "es_ES"
         LocaleCache.cache.resetCurrent(to: prefs)
         let formattedSpanish = date.formatted(.relative(presentation: .named).locale(locale))
-        
+
         // Get a formatted result from en-US
         prefs.languages = ["en-US"]
         prefs.locale = "en_US"
         LocaleCache.cache.resetCurrent(to: prefs)
         let formattedEnglish = date.formatted(.relative(presentation: .named).locale(locale))
-        
+
         // Reset to current preferences before any possibility of failing this test
         LocaleCache.cache.reset()
 
         // No matter what 'current' was before this test was run, formattedSpanish and formattedEnglish should be different.
         XCTAssertNotEqual(formattedSpanish, formattedEnglish)
     }
+
+    @available(FoundationPreview 0.4, *)
+    func testAllowedFieldsNamed() throws {
+        var named = Date.RelativeFormatStyle(presentation: .named, locale: enUSLocale, calendar: calendar)
+
+        func _verifyStyle(_ dateStr: String, relativeTo: String, expected: String, file: StaticString = #file, line: UInt = #line) {
+            let date = try! Date.ISO8601FormatStyle().parse(dateStr)
+            let refDate = try! Date.ISO8601FormatStyle().parse(relativeTo)
+            let formatted = named._format(date, refDate: refDate)
+            XCTAssertEqual(formatted, expected, file: file, line: line)
+        }
+
+        named.allowedFields = [.year]
+        _verifyStyle("2021-06-10T12:00:00Z", relativeTo: "2021-06-10T12:00:00Z", expected: "this year")
+        _verifyStyle("2020-12-06T12:00:00Z", relativeTo: "2021-01-10T12:00:00Z", expected: "last year")
+        named.allowedFields = [.year, .hour]
+        _verifyStyle("2021-06-11T12:00:00Z", relativeTo: "2021-06-01T12:00:00Z", expected: "in 240 hours")
+        _verifyStyle("2020-12-06T12:00:00Z", relativeTo: "2021-01-10T12:00:00Z", expected: "840 hours ago")
+        _verifyStyle("2020-01-10T12:00:00Z", relativeTo: "2021-01-10T12:00:00Z", expected: "last year")
+        named.allowedFields = [.minute]
+        _verifyStyle("2021-06-10T11:59:31Z", relativeTo: "2021-06-10T12:00:00Z", expected: "this minute")
+        _verifyStyle("2021-06-10T11:59:30Z", relativeTo: "2021-06-10T12:00:00Z", expected: "1 minute ago")
+        _verifyStyle("2021-06-10T11:59:00Z", relativeTo: "2021-06-10T12:00:00Z", expected: "1 minute ago")
```

Comment:
Nit: duplicated line?

Reply:
Nope, 351 is checking at 30 seconds, 352 is checking at 00 seconds.